client/src/components/PacmanGame/gameCore.js

```diff
@@ -146,34 +146,33 @@ export const getRandomAdjacentAvailableCell = (gridState, currentLocation) => {
   return getRandomAdjacentAvailableCell(gridState, currentLocation);
 };
 
-const getGridwithWeights = (grid) => {
+export const getGridwithWeights = (grid) => {
   const gridwithWeights = grid.map((array) => {
     const newArray = array.map((element) => {
       let weight = element;
+      if (weight !== 4) weight = 0;
       if (weight === 4) weight = 1;
```

Comment:
Use `entityToCode` rather than directly using the code. That will make it more readable.